docs/vanilla.md

```diff
@@ -364,6 +364,8 @@ Below is an example command of Velero restore.
 velero restore create --from-backup <your-backup-name>
 ```
 
+Note that a StorageClass with the WaitForFirstConsumer Volume Binding Mode is not supported at restore time. Only a StorageClass with the Immediate Volume Binding Mode is supported at restore time. You'll have to change the Volume Binding Mode from WaitForFirstConsumer to Immediate for restore to work.
```

Comment:
Do we need modify this for supervisor and guest?

Reply:
The Supervisor doc's Restore section points to the Vanilla doc.
For the Guest, it is more complicated as the StorageClass can't be modified (it will be synced back).  Even on Supervisor, the StorageClass is created automatically and not supposed to be modified.  So I removed "You'll have to change the Volume Binding Mode from WaitForFirstConsumer to Immediate for restore to work."